### PR TITLE
Apply w3c timeout

### PR DESCRIPTION
**Pull request by KazuCocoa**
Require adapting https://github.com/appium/appium-base-driver/pull/164 as base driver.
It means we should update the version of the driver in `package.json`

### CODE REVIEW

File: lib/commands/timeout.js [hunk 1]
```diff
@@ -1,25 +1,46 @@
...
-      this.setPageLoadTimeout(ms);
-      break;
-    case 'script':
-      this.setAsyncScriptTimeout(ms);
-      break;
-    default:
-      throw new Error(`Invalid timeout '${type}'`);
+// Arguments will be: [{"protocol":"W3C","implicit":30000}, "1dcfe021-8fc8-49bd-8dac-e986d3091b97", ...]
+// eslint-disable-next-line no-unused-vars
+commands.timeouts = async function (timeoutsObj, ...args) {
```
Comment (mykola-mokhnach) [inline lib/commands/timeout.js]: you can skip ...args declaration

File: lib/commands/timeout.js [hunk 2]
```diff
@@ -1,25 +1,46 @@
...
+
+    switch (type) {
+      case 'command':
+        this.newCommandTimeout(ms);
+        break;
+      case 'implicit':
+        this.implicitWait(ms);
+        break;
+      case 'page load':
+        this.setPageLoadTimeout(this.parseTimeoutArgument(ms));
```
Comment (mykola-mokhnach) [inline lib/commands/timeout.js]: shall we apply  `parseTimeoutArgument` to all method calls like it was in the previous implementation or to this one only?

Comment (KazuCocoa) [reply lib/commands/timeout.js]: oops, I forgot to wrap this line...

**Comment by imurchie** [comment]
Thanks for this! Published in `appium-ios-driver@1.31.0`.

**Pull request merged**